twofactor_privacyidea/lib/Provider/TwoFactorPrivacyIDEAProvider.php

```diff
@@ -349,7 +350,8 @@ public function isTwoFactorAuthEnabledForUser(IUser $user)
                 $piexcludegroupsCSV = str_replace("|", ",", $piexcludegroups);
                 $groups = explode(",", $piexcludegroupsCSV);
                 foreach($groups as $group) {
-                    if ($this->groupManager->isInGroup($user->getUID(), trim($group))) {
+                    if (($this->groupManager->isInGroup($user->getUID(), trim($group)) && $piexclude !== "0")
```

Comment:
So you are using ``piexclude !== 0`` for backward compatibility.
Would it be possible to use a default value in line 344 in getAppValue like:

   $piexclude = $this->getAppValue('piexclude', "1");

See https://doc.owncloud.org/api/classes/OCP.IConfig.html#method_getAppValue

Then we could write the comparison as

   if (($this->groupManager->isInGroup($user->getUID(), trim($group)) && $piexclude === "1")

which is I think from a logical view better readable.

Reply:
It is possible, but then we have to change it every time, cause we defined getAppValue in line 108ff without default
```php
    private function getAppValue($key)
    {
        return $this->config->getAppValue('twofactor_privacyidea', $key);
    }
```